webapp/platform/components/src/generic_modal/generic_modal.tsx

```diff
@@ -146,9 +148,11 @@ export const GenericModal: React.FC<Props> = ({
     const [showState, setShowState] = useState(show);
 
     const onHideCallback = useCallback(() => {
-        setShowState(false);
+        if (!preventClose) {
+            setShowState(false);
+        }
         onHide?.();
-    }, [onHide]);
+    }, [onHide, preventClose]);
```

Comment:
Probably there is no better way to name things, but this feels weird.

We call onHide, but nothing is really hiding (since we are preventing the close). Here some ideas, but I am really 0/5 on them, so no changes needed if you don't like them.

- Change the names to something like "tryHide"
- Make it so `onHide` returns a boolean. True means "yeah, go ahead and hide it" and false means "nope, you cannot hide it". That will make it so we don't have to pass a new prop.

Reply:
the names are more for following the RB modal names and the wrapper on it (GenericModal)